plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServer.java

```diff
@@ -53,23 +53,10 @@ public void setDefaults(IProgressMonitor monitor) {
     }
 
 	@Override
-	public URL getModuleRootURL(IModule module) {
-		IProject moduleProject = module.getProject();
-		String deployProjectName = OpenShiftServerUtils.getDeployProjectName(getServer());
-		IProject deployProject = ProjectUtils.getProject(deployProjectName);
-		String contextRoot = getContextRoot(moduleProject, deployProject);
-		return JBossServer.getModuleRootURL(module, getServer().getHost(), 80, contextRoot);
+	public URL getModuleRootURL(final IModule module) {
+		return getExtendedProperties().getModuleRootURL(module);
```

Comment:
This really shouldn't be moved. Other classes (like publishing) use these methods, and they are overriding behavior from the superclass DeployableServer.

You'd be better adapting your server to DeployableServer and using the API directly.